sequins.go

```diff
@@ -138,6 +141,31 @@ func (s *sequins) refresh() error {
 	return nil
 }
 
+func (s *sequins) download(version, destPath string) (rterr error) {
+	// To avoid loading an incomplete download (#12), download into a temp dir
+	// then rename the temp dir to destPath only if all downloads succeed.
+	baseDir := path.Dir(destPath)
+	workDir, err := ioutil.TempDir(baseDir, fmt.Sprintf("version-%v", version))
+	if err != nil {
+		return err
+	}
+	defer func() {
+		// Clean up the temp download dir in the event of a download error
+		if err := os.RemoveAll(workDir); err != nil && !os.IsNotExist(err) {
+			rterr = err
```

Comment:
This is the magical way that go lets you return values from a defer. Since the result var is named, assigning to it changes the return value. :sparkles: It's... not the best thing about the language.